providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/EventStreamObserver.java

```diff
@@ -0,0 +1,76 @@
+package dev.openfeature.contrib.providers.flagd;
+
+import java.util.Map;
+import io.grpc.stub.StreamObserver;
+import lombok.extern.slf4j.Slf4j;
+import dev.openfeature.flagd.grpc.Schema.EventStreamResponse;
+import com.google.protobuf.Value;
+
+/**
+ * EventStreamObserver handles events emitted by flagd.
+ */
+@Slf4j
+public class EventStreamObserver implements StreamObserver<EventStreamResponse> {
+    private EventStreamCallback callback;
+    private FlagdCache cache;
+
+    private static final String configurationChange = "configuration_change";
+    private static final String providerReady = "provider_ready";
+
+    EventStreamObserver(FlagdCache cache, EventStreamCallback callback) {
+        this.cache = cache;
+        this.callback = callback;
+    }
+
+    @Override
+    public void onNext(EventStreamResponse value) {
+        switch (value.getType()) {
+            case configurationChange:
+                this.handleConfigurationChangeEvent(value);
+                break;
+            case providerReady:
+                this.handleProviderReadyEvent();
+                break;
+            default:
+                log.debug("unhandled event type {}", value.getType());
+                return;
+        }
+    }
+
+    @Override
+    public void onError(Throwable t) {
+        log.error("event stream", t);
+        this.cache.clear();
+        this.callback.setEventStreamAlive(false);
+        try {
+            this.callback.restartEventStream();
+        } catch (Exception e) {
+            log.error("restart event stream", e);
+        }
+    }
+
+    @Override
+    public void onCompleted() {
+        this.cache.clear();
+        this.callback.setEventStreamAlive(false);
+    }
+
+    private void handleConfigurationChangeEvent(EventStreamResponse value) {
+        if (!this.cache.getEnabled()) {
+            return;
+        }
+
+        Map<String, Value> data = value.getData().getFieldsMap();
+        Value flagsValue = data.get("flags");
```

Comment:
@skyerus This get operation has the potential to create a NPE. Similar to https://github.com/open-feature/go-sdk-contrib/issues/84 for GO, we will need special handling here. 

Plus, let's extract `flags` to a static constant (similar to configurationChange or providerReady)

Reply:
Nice spot, thanks!
Implemented a null check, please check that I've checked sufficiently against a NPE.

Reply:
@skyerus yes, the new logic looks good and sufficiently covers the usage case.